fate/src/main/java/org/apache/accumulo/fate/zookeeper/Retry.java

```diff
@@ -125,12 +149,51 @@ public long retriesCompleted() {
   }
 
   public void waitForNextAttempt() throws InterruptedException {
-    log.debug("Sleeping for " + currentWait + "ms before retrying operation");
+    if (log.isDebugEnabled()) {
+      log.debug("Sleeping for " + currentWait + "ms before retrying operation: " + getCaller());
+    }
     sleep(currentWait);
     currentWait = Math.min(maxWait, currentWait + waitIncrement);
   }
 
   protected void sleep(long wait) throws InterruptedException {
     Thread.sleep(wait);
   }
+
+  public void logRetry(String message, Throwable t) {
+    // log the first time, and then after every logInterval
+    if (lastRetryLog < 0 || (System.currentTimeMillis() - lastRetryLog) > logInterval) {
+      log.warn(getMessage(message), t);
+      lastRetryLog = System.currentTimeMillis();
+    }
+  }
+
+  public void logRetry(String message) {
```

Comment:
Random thought, not advocating for this.  In 2.0.0-SNAP which uses java 8, could pass in a lambda that does the actual logging.  This code just decides whether or not to call it.  Not sure, but this may remove the need to inspect the stack trace to get info about caller.

Reply:
interesting thought

Reply:
Another possibility is to have the caller pass in a logger like `logRetry(Logger log, String message)`.  If this is done, would the `getCaller()` method be needed?  Also the `setLogger()` method would not be needed for testing.

Reply:
I like that